packages/core/src/ui/widgets/colorBar/ViewportColorBar.ts

```diff
@@ -0,0 +1,173 @@
+import {
+  eventTarget,
+  VolumeViewport,
+  StackViewport,
+  Types,
+  Enums,
+  utilities,
+  getEnabledElement,
+  cache,
+} from '@cornerstonejs/core';
+import { ColorBar } from './ColorBar';
+import { ColorBarProps, ColorBarVOIRange } from './types';
+
+const { Events } = Enums;
+const DEFAULT_MULTIPLIER = 4;
+
+export interface ViewportColorBarProps extends ColorBarProps {
+  element: HTMLDivElement;
+  volumeId?: string;
+}
+
+class ViewportColorBar extends ColorBar {
+  private _element: HTMLDivElement;
+  private _volumeId: string;
+
+  constructor(props: ViewportColorBarProps) {
+    super({
+      ...props,
+      range: ViewportColorBar._getRange(props.element, props.volumeId),
+      voiRange: ViewportColorBar._getVOIRange(props.element, props.volumeId),
+    });
+
+    this._element = props.element;
+    this._volumeId = props.volumeId;
+
+    this._addCornerstoneEventListener();
+  }
+
+  public get element() {
+    return this._element;
+  }
+
+  public get enabledElement() {
+    return getEnabledElement(this._element);
+  }
+
+  protected getVOIMultipliers(): [number, number] {
+    const { viewport } = this.enabledElement;
+    // const {actor: volumeActor } = viewport.getActor(this._volumeId)
+    const volume = cache.getVolume(this._volumeId);
+    const { scaling } = volume;
+    const isPreScaled = !!scaling && Object.keys(scaling).length > 0;
+    const { Modality: modality } = volume.metadata;
+
+    if (modality === 'PT') {
```

Comment:
we should extract this logic out to a utility and call it `utility.getVOIMultiplier` since I see a repetition here both in `windowLevelTool` and here